integrations/gitlab_v2/main.py

```diff
@@ -0,0 +1,157 @@
+import typing
+from enum import StrEnum
+from typing import Any
+
+from loguru import logger
+from port_ocean.context.event import event
+from port_ocean.context.ocean import ocean
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+
+from client import CREATE_UPDATE_WEBHOOK_EVENTS, DELETE_WEBHOOK_EVENTS, GitlabClient
+from integration import GitlabProjectResourceConfig
+from utils import extract_issue_payload, extract_merge_request_payload
+
+
+class InvalidTokenException(Exception):
+    ...
+
+
+class ResourceKind(StrEnum):
+    GROUP = "group"
+    PROJECT = "project"
+    MERGE_REQUEST = "merge_request"
+    ISSUE = "issue"
+
+
+class WebHookEventType(StrEnum):
+    MERGE_REQUEST = "merge_request"
+    ISSUE = "issue"
+
+
+# Listen to the start event of the integration. Called once when the integration starts.
+@ocean.on_start()
+async def on_start() -> None:
+    logger.info("Starting musah_gitlab integration")
+    tokens = ocean.integration_config["gitlab_access_tokens"]["tokens"]
```

Comment:
The intent was to add support for multiple tokens from different GitLab accounts, allowing for a seamless resync of all resources at once, rather than requiring multiple integrations. However, I can revert to supporting a single token if this approach does not align with the intended requirements. Please let me know if i should stick to just single access token for now.